src/components/navigation/navigationItem.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+  import type { SvelteHTMLElements } from 'svelte/elements'
+  import type { IconName } from '../../../icons/meta'
+  import Icon from '../icon/icon.svelte'
+
+  type Href = $$Generic<string | undefined>
+
+  type $$Props = Omit<SvelteHTMLElements['a'], 'class' | 'href'> & {
+    href: Href
+    icon?: IconName
+    isActive?: boolean
+  }
+
+  export let href: Href
+  export let icon: IconName = undefined
+  export let isActive: boolean = false
```

Comment:
Maybe this could be
```suggestion
  export let isActive?: boolean = window.location.endsWith(href)
```

or something. It'd be nice to do some rudimentary matching to make the component easier to use :smile:

Reply:
In order for this to work, though, we'd have to have a listener for the component to update when `window.location` changes in the event that we're using client-side navigation.

Reply:
Probably worth adding - I can't imagine when you wouldn't want this to work that way by default? Otherwise the parent app will need to manage it

Reply:
I'm not seeing means of doing this with VanillaJS that will work broadly enough. There's the `popstate` event, but it only triggers in specific cases, and SvelteKit's client side navigation does not trigger the event.

Reply:
I think the normal way of doing this is patching the `pushState` & `replaceState`, as well as listening to `popstate`.

```tsx

const onChange = () => {/* do stuff*}

['pushState', 'replaceState'].forEach((name) => {
  const original = history[name]
  history[name] = function() {
    original.apply(history, arguments)
    onChange()
  }
})
```

Reply:
But that would be up to the consuming app, and not Nala, yes?

Reply:
I mean, there's not really any reason we can't do it in Nala is there? Externally (i.e. in the consuming app) nothing will change except the NavigationLinks will Just Work(tm).

Not super important, so we can leave it out if you think it's going to make this PR too complicated.